### PR TITLE
docs(self-managed): add platform architecture section and diagrams

### PR DESCRIPTION
**Pull request by aabouzaid**
## What is the purpose of the change

Adding a new section for architecture that provides a high-level overview of C8SM.

## Are there related marketing activities

No.

## When should this change go live?

ASAP.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory.
- [x] My changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer.

## To-Do
- [x] Move the content to v8.1 dir once the review is done to avoid reviewing the same content twice.

### CODE REVIEW

File: docs/self-managed/platform-architecture/overview.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Architecture
+
+Camunda Platform 8 Self-Managed has multiple web applications and gRPC services which could be accessed externally using Ingress. The following diagram is the architecture with two Ingress objects, an Ingress with HTTP(S) protocol for all web applications using a single domain, and another Ingress with gRPC protocol for the Zeebe workflow engine:
+
+![Camunda Platform 8 Self-Managed Architecture Diagram - Combined Ingress](./assets/camunda-platform-8-self-managed-architecture-diagram-combined-ingress.png)
+
+In this architecture, Camunda Platform 8 Self-Managed can be accessed as follows:
+
+- Web applications: https://camunda.example.com/[identity|operate|optimize|tasklist]
```
Comment (christinaausley) [inline docs/self-managed/platform-architecture/overview.md:18]: Non-blocking, but I wonder if there is a way we can turn this text into a link like the others below?

**Review by christinaausley — DISMISSED** [review]
A few adjustments made in my latest commit -- let me know if you have any questions 👍

File: docs/self-managed/platform-architecture/overview.md (hunk 1 above)
Comment (aabouzaid) [reply docs/self-managed/platform-architecture/overview.md:18]: The points here are not meant to be links but should be shown as they are (as an example).
Maybe we make them like this:

- Web applications: `https://camunda.example.com/[identity|operate|optimize|tasklist]`

Comment (christinaausley) [reply docs/self-managed/platform-architecture/overview.md:18]: OH -- I see what you're saying. I'll revert the others 👍

**Review by christinaausley — DISMISSED** [review]

**Comment by aabouzaid** [comment]
also 1 more question, please.
is there an automated way to apply the latest changes to versioned docs?
or it's done manually? (that is what I did for this PR).

**Review by christinaausley — APPROVED** [review]

**Comment by christinaausley** [comment]
> also 1 more question, please. is there an automated way to apply the latest changes to versioned docs? or it's done manually? (that is what I did for this PR).

I currently also manually change versions of older docs 👍

**Pull request merged**